theano/sandbox/gpuarray/tests/test_nnet.py

```diff
@@ -209,11 +207,7 @@ def softmax_with_bias_unittest_template(dtypeInput, dtypeBias):
                       GpuSoftmaxWithBias)
 
     def cmp(n, m):
-        # print "test_softmax",n,m
-        if dtypeInput == 'float32':
-            data = numpy.arange(n * m, dtype='float32').reshape(n, m)
-        elif dtypeInput == 'float64':
-            data = numpy.arange(n * m, dtype='float64').reshape(n, m)
+        data = numpy.arange(n * m, dtype=dtypeInput).reshape(n, m)
 
         out = f(data)
         gout = f_gpu(data)
```

Comment:
The assert bellow is failing as the comparison is too strict. Use:

theano.tests.unittest_tools.assert_allclose(out, gout)